__tests__/jestGlobalSetup.ts

```diff
@@ -0,0 +1,255 @@
+  }
+  if (!process.env.TEST_ACCOUNT_PRIVATE_KEY) {
+    if (final) throw new Error('TEST_ACCOUNT_PRIVATE_KEY env is not provided');
+    else warnings.push('TEST_ACCOUNT_PRIVATE_KEY env is not provided!');
+  }
+  if (!process.env.TEST_PROVIDER_BASE_URL && !process.env.TEST_RPC_URL) {
+    if (final) throw new Error('One of TEST_PROVIDER_BASE_URL or TEST_RPC_URL env is not provided');
+    else warnings.push('TEST_PROVIDER_BASE_URL or TEST_RPC_URL env is not provided!');
+  }
+
+  if (warnings.length > 0) {
+    console.log('\x1b[33m', warnings.join('\n'), '\x1b[0m');
+    delete process.env.TEST_ACCOUNT_ADDRESS;
+    delete process.env.TEST_ACCOUNT_PRIVATE_KEY;
+    delete process.env.TEST_PROVIDER_BASE_URL;
+    return false;
+  }
+
+  if (!final) {
+    overrideOrSet('IS_LOCALHOST_DEVNET', 'false');
+    overrideOrSet('IS_RPC_DEVNET', 'false');
+    overrideOrSet('IS_SEQUENCER_DEVNET', 'false');
+    overrideOrSet('IS_RPC', process.env.TEST_RPC_URL ? 'true' : 'false');
+    overrideOrSet('IS_SEQUENCER', process.env.TEST_PROVIDER_BASE_URL ? 'true' : 'false');
+    overrideOrSet(
+      'IS_SEQUENCER_GOERLI',
+      (process.env.TEST_PROVIDER_BASE_URL || process.env.TEST_RPC_URL || '').includes(
+        BaseUrl.SN_GOERLI
+      )
+        ? 'true'
+        : 'false'
+    );
+  }
+
+  console.table({
+    TEST_ACCOUNT_ADDRESS: process.env.TEST_ACCOUNT_ADDRESS,
+    TEST_ACCOUNT_PRIVATE_KEY: '****',
+    INITIAL_BALANCE: process.env.INITIAL_BALANCE,
+    TEST_PROVIDER_BASE_URL: process.env.TEST_PROVIDER_BASE_URL,
+    TEST_RPC_URL: process.env.TEST_RPC_URL,
+  });
+
+  console.table({
+    IS_LOCALHOST_DEVNET: process.env.IS_LOCALHOST_DEVNET,
+    IS_RPC_DEVNET: process.env.IS_RPC_DEVNET,
+    IS_SEQUENCER_DEVNET: process.env.IS_SEQUENCER_DEVNET,
+    IS_RPC: process.env.IS_RPC,
+    IS_SEQUENCER: process.env.IS_SEQUENCER,
+    IS_SEQUENCER_GOERLI: process.env.IS_SEQUENCER_GOERLI,
+  });
+
+  console.log('Global Test Environment is Ready');
+  return true;
+};
+
+const executeStrategy = async () => {
+  // 1. Assume setup is provided and ready;
+  console.log('Global Test Setup Started');
+  let ready = verifySetup();
+  if (ready) {
+    console.log('Using Provided Test Setup');
+    return true;
+  }
+
+  // 2. Try to detect setup
+  console.log('Basic test parameters are missing, Auto Setup Started');
+  const devnetStrategy = await localDevnetDetectionStrategy();
+  if (devnetStrategy.isDevnet) {
+    if (devnetStrategy.isRS) {
+      console.log('Detected Devnet-RS');
+    } else {
+      console.log('Detected Devnet-PY');
+    }
+  }
+
+  const providerType = await sequencerOrRpc(devnetStrategy);
+  if (providerType.sequencer) {
+    console.log('Detected Sequencer');
+  } else if (providerType.rpc) {
+    console.log('Detected RPC');
+  }
+
+  const isAccountSet = await setAccount(devnetStrategy);
+  if (isAccountSet) {
+    console.log('Detected Account');
+  }
+
+  ready = await verifySetup(true);
```

Comment:
`verifySetup` is sync so could be:
```suggestion
  ready = verifySetup(true);
```
or:
```suggestion
  return verifySetup(true);
```